File: 0179-largest-number/0179-largest-number.cpp
bool compare(string a, string b){
        return a + b > b + a;
    }

class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> string_nums;
        for(int n:nums)
            string_nums.push_back(to_string(n));
        
        sort(begin(string_nums), end(string_nums), compare);
        
        string ans;
        for(string s: string_nums)
            ans+= s;
        
        return ans[0]=='0'? "0": ans;
    }
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        for i, n in enumerate(nums):
            nums[i] = str(n)
            
        
        def compare(n1, n2):
            if n1 + n2 > n2 + n1:
                return -1
            else:
                return 1
        
        nums = sorted(nums, key = cmp_to_key(compare))
        # if we have [0,0,0] the result will be "000" but we want "0" 
        # so we convert "000" to integer so it becomes 0 then change it again to string to be "0"
        return str(int("".join(nums)))